### PR TITLE
MongoDB Requested Connection Changes

### PR DESCRIPTION
**Pull request by LukarioMC**
# Overview
Made these changes according to feedback given in Discord + extra's that Susan requested for issue #6 
## New Changes
- Added `helmet` middleware for increased security headers
- Added `compression` middleware for handling compressed content
- Added `cors` middleware for handling CORS 🥲
- Added the use of express 4.16+ `json` and `urlencoded` built-in middleware
- Moved database connection logic into the `app.listen` callback
## Changes that were already approved and merged into main (missing from dev)
- Added `dotenv` for loading environment variables
- Added `mongoose` for connecting MongoDB cluster

### CODE REVIEW

**Comment by LukarioMC** [comment]
Sorry I assigned you as reviewer because you had requested these specific changes 😄

File: apps/augury-backend/src/main.ts [hunk 1]
```diff
@@ -3,19 +3,50 @@
...
   res.send({ message: 'Welcome to augury-backend!' });
 });
 
 const port = process.env.PORT || 3333;
 const server = app.listen(port, () => {
+  mongoose.set('strictQuery', false);
+  mongoose
+    .connect(`${process.env.MONGO_URL}`)
+    .catch(() =>
+      console.error("Please configure your environment's MONGO_URL!")
```
Comment (susansli) [inline apps/augury-backend/src/main.ts:39]: Please add a semicolon :D

File: apps/augury-backend/src/main.ts [hunk 2]
```diff
@@ -3,19 +3,50 @@
...
+    );
+
+  const db = mongoose.connection;
+  db.on('error', () =>
+    console.error('Could not connect to Mongo - restart the server.')
```
Comment (susansli) [inline apps/augury-backend/src/main.ts:44]: Please add a semicolon :D

**Review by susansli — COMMENTED** [review]
Looks great, please add those semicolons and you're good! Also, thanks for writing the changes you didn, very helpful, thanks!

File: apps/augury-backend/src/main.ts (hunk 1 above)
Comment (LukarioMC) [reply apps/augury-backend/src/main.ts]: These weren't in brackets so it wasn't required, so I've changed both so they are consistent and easy to expand later w/o changing more lines 😀

File: apps/augury-backend/src/main.ts (hunk 2 above)
Comment (LukarioMC) [reply apps/augury-backend/src/main.ts]: Same response as above, should be fixed 👍

**Review by susansli — APPROVED** [review]
Oh sick, thanks 🙏✨️

**Pull request merged**